### DISCUSSION TITLE
Shutting down Windows 10

### DISCUSSION THREAD

**Discussion by ChopsKingsland** [AlexGustafsson/homebridge-wol#133] [Questions]
I want to shut down my windows 10 PC, and I followed the tutorial in the wiki, but it got stuck at copying over the generated key to my windows machine. I have come up with a new solution, though, `sshpass -p '<win-password>' ssh <win-user>@<win-pwd> 'shutdown /s'`. This works in terminal on Homebridge, but throws an error when using it in the accessory. Any ideas why?

**Comment by AlexGustafsson** [comment]
It could be any number of reasons. Can you post the verbose logs where you can see the error of the command?

**Reply by ChopsKingsland** [reply]
How do I get verbose logs?

**Reply by AlexGustafsson** [reply]
There's some documentation available in the README. Note however, that it is for v5. If you have v4 you'll want to look there instead: https://github.com/AlexGustafsson/homebridge-wol/tree/cfed7dbee3e0fde413c67ab5b9d42ec276d85556.

For the fourth version, you're looking for the following values: `log`, `logPinger`, `debugLog`. You can also start Homebridge using `DEBUG=* homebridge ...` to get even more verbose logs from homebridge as well.

**Reply by ChopsKingsland** [reply]
![image](https://user-images.githubusercontent.com/18499854/102123455-c4779d80-3e3e-11eb-8d96-0690f07342d7.png)

**Reply by AlexGustafsson** [reply]
So sshpass exits with code 6. Have you read this https://unix.stackexchange.com/questions/323730/sshpass-failing-with-exit-code-6 ?

**Reply by ChopsKingsland** [reply]
The answer on that link is for using 3 devices, so how would I do it for 1?

**Reply by AlexGustafsson** [reply]
The takeaway from the answer posted is to ensure that you've actually connected once before as the user running homebridge. It will have to have trust established in the remote device. So as the user running homebridge, run SSH once and see if you get a prompt about establishing trust.

**Reply by AlexGustafsson** [reply]
Were you able to solve the issue?

**Comment by bedotousa** [comment]
Why do i keep getting the mails? I have unsubscribed and it keeps coming..

Sendt fra min iPhone

**Reply by ChopsKingsland** [reply]
are you watching the repo?

**Reply by AlexGustafsson** [reply]
Yes, you're "watching" this repository so therefore you'll be notified of all updates: https://github.com/AlexGustafsson/homebridge-wol/watchers. You can select to only be notified after new releases.

**Comment by ropi59** [comment]
hi! i'm new here. i have same problem to shutdown pc on win 10. i follow all steps on tuto but the shutdown command won't work. any help please? (apologize for my english i'm french...^^

![logs homebridge](https://user-images.githubusercontent.com/58685834/111317229-3e990780-8664-11eb-80a9-e94150d5709b.png)
![config homebridge](https://user-images.githubusercontent.com/58685834/111317237-4193f800-8664-11eb-934b-d81959b0ff63.png)

)

**Reply by AlexGustafsson** [reply]
You're getting "permission denied" when trying SSH into the machine. That may be for any number of reason. For example, your username (`arret_pc`) may be wrong, your SSH key `/homebridge/id_rsa` might not exist or it might not be added to your PC as a trusted key yet.

1. Ensure that `arret_pc` is the correct username and that you can SSH into the machine by running `ssh arret_pc@192.168.0.19` on the machine running homebridge
2. Ensure that your SSH key exists: `cat /homebridge/id_rsa`
3. Try to SSH into the machine using that key: `ssh -i /homebridge/id_rsa arret_pc@192.168.0.19`

If step 1 fails, you know where to look.

If step 2 fails, you will need to generate a SSH key.

If step three fails but step one worked, you'll need to add the key to the remote PC:

```
ssh-copy-id -i /homebridge/id_rsa arret_pc@192.168.0.19
```

**Reply by ropi59** [reply]
thanks for your rapid response! wich username do i use? arret_pc is a user i create on windows 10 but IDK if it's the username homebridge ask...
on step 1 and step 3 homebridge ask for a password. where could i found it? it was my session password on windows?

**Reply by AlexGustafsson** [reply]
The username you should use is the username you've created on the target machine. In your case, `arret_pc` seems correct.

The password you should enter is the password for the user `arret_pc` on the target computer.

**Reply by ropi59** [reply]
the password i need is the session password on windows? and if my session was named firstname name i have a probleme on homebridge. if i create user and i use another user the shutdown will run?
sorry for all this questions i discover homebridge... ^^

**Reply by ropi59** [reply]
i make a mistake on your tutorial. i juste remove extension to authorized_keys (before it was authorized_keys.txt now just authorized_keys) but now my logs affraid me!!!!
![logs homebridge](https://user-images.githubusercontent.com/58685834/111332736-56778800-8672-11eb-91af-19ba87907102.png)

**Reply by ropi59** [reply]
i have uninstall homebridge and reinstall it and i have the same problem. do you know how i can resolve it? thanks

**Reply by AlexGustafsson** [reply]
This error is quite common. It states that the machine you're trying to connect to no longer uses the same key you have marked as trusted. This can happen if the machine is reinstalled or has a new key setup for SSH. If you trust the machine (since this is your local network, you probably do), you will have to edit `/root/.ssh/known_hosts` and remove line 1. This basically marks the machine as untrusted, so that you can repeat the steps again.

**Reply by ropi59** [reply]
where is this file because IDK where found root/.ssh/known_hosts?

**Reply by ropi59** [reply]
is it on my nas or my windows pc?

**Reply by AlexGustafsson** [reply]
The file is in `/root/.ssh/known_hosts`. It will be on your NAS. For simplicity's sake you may execute `sudo rm /root/.ssh/known_hosts`. This will remove _all_ trusted hosts.

**Reply by ropi59** [reply]
now the error disapear but it ask me for a password to connect in ssh... i'm going insane! ^^

**Reply by ropi59** [reply]
![Sans titre](https://user-images.githubusercontent.com/58685834/111483943-93588300-8735-11eb-9711-6923a170cfbd.png)
WOL works perfectly but IDK how resolve my problem for shutdown...

**Reply by ropi59** [reply]
here is my logs on homebridge
![Sans titre1](https://user-images.githubusercontent.com/58685834/111484493-1b3e8d00-8736-11eb-985b-b6415c6ee3cf.png)

**Reply by AlexGustafsson** [reply]
Rinse and repeat. Your issue is the same as the one you posted about first. See the solution here: https://github.com/AlexGustafsson/homebridge-wol/discussions/133#discussioncomment-488780.

**Reply by ropi59** [reply]
in homebridge terminal i execute sudo rm /root/.ssh/known_hosts. then i execute ssh-keygen part on your tutorial. i comment last lines in sshd_config with #.
i copy id_rsa.pub and rename it on authorized_keys (with no extension).
i disable inheritance except for user and system and restart open ssh ssh server.
then when i execute ssh -o StrictHostKeyChecking=no -i /homebridge/id_rsa username@remote_Address shutdown /s i have this:
![Sans titre](https://user-images.githubusercontent.com/58685834/111506452-edfbda00-8749-11eb-824c-9d2144a6e86b.png)

**Reply by ropi59** [reply]
i have retry 3 times at least all the tuto. it works one time but when i'd restart my computer i have the remote host identification warning and i won't able to shutdown anymore. 
i try another time to do the tuto. i'm connect in ssh with the command 
ssh -i /homebridge/id_rsa username@remote_Address
but the command to shutdown don't work without password. (and i haven't this f**** password! ^^)
![1](https://user-images.githubusercontent.com/58685834/111516290-0670f200-8754-11eb-9e95-41c032c30b4f.png)

**Reply by AlexGustafsson** [reply]
In your screenshot it says that `/homebridge/id_rsa` is not accessible. That corresponds to step 2 from my previously mentioned comment. You will have to create a new key.

**Reply by ropi59** [reply]
now it works in homebridge terminal. but it doesn't works with interuptor in apple's home app...

**Reply by ropi59** [reply]
when i add homebridge to home app i have error warning remote identification has changed. WOL works perfectly. i think my line to shutdown is wrong in homebridge config:
"shutdownCommand": "ssh -o StrictHostKeyChecking=no -i /homebridge/id_rsa olivier@192.168.0.19 shutdown /s",

**Reply by AlexGustafsson** [reply]
If you have a warning about the remote identification having changed, the issue is the same as the one previously outlined. Basically, the user running homebridge (could be `root`, `homebridge` or anything), has a file which marks the remote PC's key as trusted. The full error will contain the path to the file and the line in the file that you may have to remove in order to trust the machine anew.

Your command looks good, but likely, another key is already marked as trusted for the PC, stopping `StrictHostKeyChecking=no` from working.

**Reply by ropi59** [reply]
IDK if it's a good solution but i remove SrictHostKeyChecking=no from my shutdown command and i use chmod /homebridge/id_rsa in homebridge Terminal and now all works perfectly. Thanks for your help!!!!!

**Reply by ropi59** [reply]
So i'm really bad with this function. it works one time and now:
![1](https://user-images.githubusercontent.com/58685834/111779874-f160a400-88b6-11eb-99f5-6fe59560f408.png)
i'm going crazy!!!!

**Reply by AlexGustafsson** [reply]
Remove the `-o`.
